test/core/dictionary.cpp

```diff
@@ -68,4 +68,12 @@ TEST_CASE("Dictionary operations", "[dictionary]")
         NeoFOAM::Dictionary& sDict2 = dict.subDict("subDict");
         REQUIRE(sDict2.get<int>("key1") == 100);
     }
+
+    SECTION("initialize with map")
+    {
+        NeoFOAM::Dictionary dict_init({{"key1", 42}, {"key2", std::string("Hello")}});
```

Comment:
Use camelCase instead of snake_case
```suggestion
        NeoFOAM::Dictionary dictInit({{"key1", 42}, {"key2", std::string("Hello")}});
```